heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -0,0 +1,426 @@
+        '$.reporters.*[?(@property == 0 && @ !== "default")]': nodeResolveMetadata, // First entry in [ path, options ], excluding "default"
+        // watchPlugins: (path | [ path, options ])[]
+        '$.watchPlugins.*@string()': nodeResolveMetadata, // string path
+        '$.watchPlugins.*[?(@property == 0)]': nodeResolveMetadata, // First entry in [ path, options ]
+        // transform: { [regex]: path | [ path, options ] }
+        '$.transform.*@string()': nodeResolveMetadata, // string path
+        '$.transform.*[?(@property == 0)]': nodeResolveMetadata // First entry in [ path, options ]
+      }
+    });
+  }
+
+  public apply(
+    heftSession: HeftSession,
+    heftConfiguration: HeftConfiguration,
+    options?: IJestPluginOptions
+  ): void {
+    // TODO: Remove when newer version of Heft consumed
+    if (options) {
+      JsonSchema.fromFile(SCHEMA_PATH).validateObject(options, 'config/heft.json');
+    }
+
+    heftSession.hooks.build.tap(PLUGIN_NAME, (build: IBuildStageContext) => {
+      build.hooks.compile.tap(PLUGIN_NAME, (compile: ICompileSubstage) => {
+        compile.hooks.afterCompile.tapPromise(PLUGIN_NAME, async () => {
+          // Write the data file used by jest-build-transform
+          if (build.properties.isTypeScriptProject) {
+            await JestTypeScriptDataFile.saveForProjectAsync(heftConfiguration.buildFolder, {
+              emitFolderNameForTests: build.properties.emitFolderNameForTests || 'lib',
+              extensionForTests: build.properties.emitExtensionForTests || '.js',
+              skipTimestampCheck: !build.properties.watchMode
+            });
+          }
+        });
+      });
+    });
+
+    heftSession.hooks.test.tap(PLUGIN_NAME, (test: ITestStageContext) => {
+      test.hooks.run.tapPromise(PLUGIN_NAME, async () => {
+        await this._runJestAsync(heftSession, heftConfiguration, test, options);
+      });
+    });
+
+    heftSession.hooks.clean.tap(PLUGIN_NAME, (clean: ICleanStageContext) => {
+      this._includeJestCacheWhenCleaning(heftConfiguration, clean);
+    });
+  }
+
+  private async _runJestAsync(
+    heftSession: HeftSession,
+    heftConfiguration: HeftConfiguration,
+    test: ITestStageContext,
+    options?: IJestPluginOptions
+  ): Promise<void> {
+    const jestLogger: ScopedLogger = heftSession.requestScopedLogger('jest');
+    const buildFolder: string = heftConfiguration.buildFolder;
+
+    this._jestTerminal = jestLogger.terminal;
+    this._jestTerminal.writeLine(`Using Jest version ${getVersion()}`);
+
+    // In watch mode, Jest starts up in parallel with the compiler, so there's no
+    // guarantee that the output files would have been written yet.
+    if (!test.properties.watchMode) {
+      this._validateJestTypeScriptDataFile(buildFolder);
+    }
+
+    let jestConfig: string;
+    if (options?.resolveConfigurationModules === false) {
```

Comment:
Defaults to true because I want the feature to be enabled by default. The other way was confusing. Any ideas on what it could be named that implies that enabled-by-default is the intended purpose?

Reply:
Talked offline, resolved